homeassistant/components/google_travel_time/sensor.py

```diff
@@ -63,7 +66,10 @@ async def async_setup_entry(
         options = new_data.pop(CONF_OPTIONS, {})
 
         if CONF_UNITS not in options:
-            options[CONF_UNITS] = hass.config.units.name
+            if hass.config.units.length_unit == LENGTH_MILES:
+                options[CONF_UNITS] = UNITS_IMPERIAL
+            else:
+                options[CONF_UNITS] = UNITS_METRIC
```

Comment:
Checking/Storing the `name` of the unit system will stop working if we rename or add a new unit system.
It is more future-proof to check the `length_unit` of the unit system, even if the backend library still uses "imperial" terminology.